src/astro/files/base.py

```diff
@@ -0,0 +1,80 @@
+import os
+import pathlib
+from typing import List, Optional
+
+import pandas as pd
+import smart_open
+
+from astro.constants import FileType
+from astro.files.locations import create_file_location
+from astro.files.type import create_type_factory
+
+
+class File:
+    def __init__(self, path: str, conn_id: Optional[str] = None):
+        """Init file object which represent a single file in local/object stores
+
+        :param path: Path to a file in the filesystem/Object stores
+        :param conn_id: Airflow connection ID
+        """
+        self.location = create_file_location(path, conn_id)
+        self.type = create_type_factory(path)
+
+    @property
+    def path(self):
+        return self.location.path
+
+    def get_size(self) -> int:
+        """
+        Return the size (bytes) of the given file.
+
+        :return: File size in bytes
+        :rtype: int
+        """
+        path = pathlib.Path(self.path)
+        return os.path.getsize(path)
```

Comment:
I strongly suggest this method is implemented per FileLocation, since the current implementation only works for local files.

Reply:
done.

Reply:
Should we change this method to use the `FileLocation` implementation of `get_size`?

Reply:
I have moved this method to respective location classes, with -1 returning for locations that are nonlocal.